Assets/MixedRealityToolkit.Extensions/SceneTransitionService/Scripts/CameraFaderQuad.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.SceneTransitions
+{
+    public class CameraFaderQuad : ICameraFader
+    {
+        const string QuadMaterialShaderName = "UI/Unlit/Transparent";
+        const string QuadMaterialColorName = "_Color";
+
+        /// <summary>
+        /// Simple struct for keeping track of quad properties
+        /// </summary>
+        private struct Quad
+        {
+            public Renderer Renderer;
+            // Eventually we want to be able to have different quads use different colors
+            // Using property blocks now keeps our options open
+            public MaterialPropertyBlock PropertyBlock;
+        }
+
+        public CameraFaderState State { get; private set; }
+
+        private Dictionary<Camera, Quad> quads = new Dictionary<Camera, Quad>();
+        private Color fadeOutColor;
+        private Color fadeInColor;
+        private Color currentColor;
+        private Material quadMaterial;
+
+        public async Task FadeOutAsync(float fadeOutTime, Color color, IEnumerable<Camera> targets)
+        {
+            switch (State)
+            {
+                case CameraFaderState.Clear:
+                    break;
+
+                default:
+                    Debug.LogWarning("Can't fade out in state " + State + " - not proceeding.");
+                    return;
+            }
+
+            State = CameraFaderState.FadingOut;
+
+            fadeOutColor = color;
+            fadeInColor = fadeOutColor;
+            fadeInColor.a = 0;
+
+            if (fadeOutColor.a < 1)
+            {
+                Debug.LogWarning("Target color is not fully opaque.");
+            }
+
+            await Task.Yield();
```

Comment:
>await Task.Yield(); [](start = 12, length = 19)

What's the significance of this particular Task.Yield?

Reply:
I think it was a remnant from an earlier version. Removed.